docs/source/projects.rst

```diff
@@ -56,24 +55,106 @@ project for remote execution on `Databricks <https://databricks.com>`_.
 Specifying Projects
 -------------------
 
-By default, any Git repository or local directory is treated as a project, and MLflow uses the
-following conventions to determine its parameters:
+By default, any Git repository or local directory can be treated as an MLflow project; you can
+invoke any bash or Python script contained in the directory as a project entry point. The 
+:ref:`project-directories` section describes how MLflow interprets directories as projects.
+
+To provide additional control over a project's attributes, you can also include an :ref:`MLProject
+file <mlproject-file>` in your project's repository or directory.
+
+Finally, MLflow projects allow you to specify the software :ref:`environment <project-environments>`
+that is used to execute project entry points.
+
+.. _project-environments:
+
+Project Environments
+^^^^^^^^^^^^^^^^^^^^
+MLflow currently supports the following project environments: Conda environment, Docker container, and system environment.
+
+.. _project-conda-environments:
+
+Conda environment
+  `Conda <https://conda.io/docs>`_ environments support 
+  both Python packages and native libraries (e.g, CuDNN or Intel MKL). When an MLflow Project 
+  specifies a Conda environment, it is activated before project code is run.
+
+  By default, MLflow uses the system path to find and run the ``conda`` binary. You can use a 
+  different Conda installation by setting the ``MLFLOW_CONDA_HOME`` environment variable; in this 
+  case, MLflow attempts to run the binary at ``$MLFLOW_CONDA_HOME/bin/conda``.
+
+  You can specify a Conda environment for your MLflow project by including a ``conda.yaml``
+  file in the root of the project directory, or by including a ``conda_env`` entry in your
+  ``MLProject`` file. For more information, see the :ref:`project-directories` and 
+  :ref:`MLProject File <mlproject-file>` sections.
+
+.. _project-docker-container-environments:
+
+Docker container
+  `Docker containers <https://www.docker.com/resources/what-container>`_ allow you to 
+  capture non-Python dependencies such as Java libraries. When you run an MLflow Project that 
+  specifies a Docker image, MLflow runs the image, mounts the project directory in the resulting 
```

Comment:
We actually build a new image based on the project-specified image with the project directory COPY'd in to `/mlflow/projects/code`, so maybe we can reword to something like:

```
When you run an MLflow Project that specifies a Docker image IMG, MLflow builds a new image based on IMG with the project directory copied into `/mlflow/projects/code`. It then runs the image, invoking the project entry point in the resulting container.
```

Reply:
@smurching Thanks for the clarification! I've updated the phrasing to indicating that the directory gets copied. Let me know if this looks okay!